### ISSUE TITLE
Creation of Architectural Design Documents

### ISSUE THREAD

**Issue by FendVario1** [amosproj/amos2022ss01-firmware-downloader#10]
## User story
1. As a Developer
2. I want / need a general software-architectural design
3. So that it is possible for all developers to follow this design.

## Acceptance criteria
* A general architectural design, all SDs agreed on within a respective document.
* SDs and POs not fully agreed on model, so further updates are needed
* Modularized Architecture (per vendor download modules) should be represented

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
